### PR TITLE
Add tag and annotation to OneRegionTimeseriesDataset

### PR DESCRIPTION
**Pull request by TomGoBravo**
This PR is part of https://trello.com/c/Cq2zNk2M/674-annotate-data-outliers-that-weve-flagged-in-the-ui

* Copies all tags associated with a location_id to `OneRegionTimeseriesDataset` when creating it
* Makes `OneRegionTimeseriesDataset.provenance` from the tag DataFrame
* Adds `TagInTimeseries`, a frozen dataclass to represent a tag in the context of a particular timeseries. Use it in production code to iterate through tags and to replace named tuple `AnnotationInTimeseriesLiteral`. Unfortunately it is harder to make a literal dataclass than a tuple so adds `make_tag` to simplify making them in tests.
* Extracts code that turns a list of tags to a DataFrame to `make_tag_df` where it can be reused by a test calling `append_tag_df`

### CODE REVIEW

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -126,13 +136,28 @@ class OneRegionTimeseriesDataset:
...
-    provenance: Dict[str, str] = dataclasses.field(default_factory=dict)
+    # A default exists for convenience in tests. Non-test code is expected to explicitly set tag.
+    tag: pd.Series = dataclasses.field(
+        default_factory=lambda: _EMPTY_TAG_SERIES.reset_index(CommonFields.LOCATION_ID, drop=True)
+    )
 
-    # TODO(tom): find a way to expose all tags in this class or maybe add `attributes` as tags with
-    # a real date.
+    @property
+    def provenance(self) -> Dict[str, str]:
```
Comment (ghop02) [inline libs/datasets/timeseries.py:145]: I think eventually this could return a typed pydantic object (that shows up in the api)

**Review by ghop02 — APPROVED** [review]
looks good to me!  Couple small comments, but nothing blocking

File: libs/datasets/timeseries.py [hunk 2]
```diff
@@ -114,6 +114,16 @@ class RegionLatestNotFound(IndexError):
     pass
 
 
+@final
+@dataclass(frozen=True)
+class TagInTimeseries:
```
[the remark this answers is not in the document]
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:119]: Yeah, not a great name but I'm trying to be clear that this is only meaningful in the context of a particular timeseries. TimeseriesTag is more likely to include something that identifies the timeseries.

File: libs/datasets/timeseries.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:145]: It can be whatever you want, as long as it is a read-only view of `tag` :-P

**Pull request merged**